packages/automerge-repo/src/DocHandle.ts

```diff
@@ -1,181 +1,211 @@
+   * be visible to the user. A patch is the result of one or more changes. It describes the
+   * difference between the state before and after the changes.
+   */
+  #emitPatch(
     patch: any, //Automerge.Patch,
-    before: Automerge.Doc<T>,
-    after: Automerge.Doc<T>
+    before: A.Doc<T>,
+    after: A.Doc<T>
   ) {
     this.emit("patch", { handle: this, patch, before, after })
   }
 
+  /**
+   * This is the current state of the document. If a document isn't available locally, this will
+   * block until until we get it from a peer. (As noted above, we should probably have a timeout.)
+   */
   async value() {
-    if (!this.ready()) {
-      log(`[${this.documentId}]: value: (${this.state}, waiting for ready)`)
-      await new Promise((resolve) => this.once("ready", () => resolve(true)))
+    if (!this.isReady()) {
+      this.#log(
+        `[${this.documentId}]: value: (${this.state}, waiting for ready)`
+      )
+      await new Promise(resolve => this.once("ready", () => resolve(true)))
     } else {
-      await new Promise((resolve) => setTimeout(() => resolve(true), 0))
+      await pause(0)
     }
-    log(`[${this.documentId}]: value:`, this.doc)
+    this.#log(`[${this.documentId}]: value:`, this.doc)
     return this.doc
   }
 
-  // this returns the value but not until after loading is done
-  // but it will return the value during syncing when we don't want to share it
-  // with the frontend / user code
-  async syncValue() {
-    log(`[${this.documentId}]: syncValue,`, this.doc)
-    if (this.state == "loading") {
-      log(`[${this.documentId}]: value: (${this.state}, waiting for syncing)`)
-      await new Promise((resolve) => {
-        this.once("syncing", () => resolve(true))
+  /**
+   * If a document isn't available locally, this will return an empty document while we're asking
+   * peers for it.
+   */
+  async provisionalValue() {
+    this.#log(`[${this.documentId}]: provisionalValue,`, this.doc)
+    if (this.state == HandleState.LOADING) {
+      this.#log(
+        `[${this.documentId}]: value: (${this.state}, waiting for syncing)`
+      )
+      await new Promise(resolve => {
+        this.once("requesting", () => resolve(true))
         this.once("ready", () => resolve(true))
       })
     } else {
-      await new Promise((resolve) => setTimeout(() => resolve(true), 0))
+      await pause(0)
     }
-    log(`[${this.documentId}]: syncValue:`, this.doc)
+    this.#log(`[${this.documentId}]: provisionalValue:`, this.doc)
     return this.doc
   }
 
-  change(callback: (doc: T) => void, options: ChangeOptions<T> = {}) {
-    this.value().then(() => {
-      const newDoc = Automerge.change<T>(this.doc, options, callback)
-      this.__notifyChangeListeners(newDoc)
-    })
+  /** Applies an Automerge change function to the document. */
+  async change(callback: A.ChangeFn<T>, options: ChangeOptions<T> = {}) {
+    await this.value()
```

Comment:
we should note that this is blocking to make sure you don't call change() before you get an initial value by accident